src/main/scala/dpla/ingestion3/harvesters/resourceSync/RsRelation.scala

```diff
@@ -0,0 +1,81 @@
+package dpla.ingestion3.harvesters.resourceSync
+
+import org.apache.spark.sql._
+import org.apache.spark.sql.sources._
+import org.apache.spark.sql.types._
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.catalyst.ScalaReflection
+
+/**
+  * This class interprets the user-submitted params to determine which type of
+  * request to send to the ResourceSync feed (ie. get all records or a change set,
+  * get those records by iterating over resourcelist or requesting TAR files via dump).
+  * It requests a ResourceSync response via the `RsResponseBuilder`.
+  * It constructs a DataFrame from the RS response.
+  */
+class RsRelation (endpoint: String,
+                  harvestAll: Boolean = true,
+                  capability: String = "resourcelist")
+                 (@transient val sqlContext: SQLContext)
+  extends BaseRelation with TableScan with Serializable {
+
+  val rsResponseBuilder = new RsResponseBuilder(endpoint)(sqlContext)
+
+  /**
+    * Make appropriate call to RsResponseBuilder
+    */
+  def records: RDD[RsResponse] = {
+    (capability, harvestAll) match {
+        case ("resourcelist", true) => rsResponseBuilder.getAllResourceList()
+        case ("resourcelist", false) => ???
+        case ("resourcedump", true) => ???
+        case ("resoucedump", false) => ???
+        // Resources can be described by as sitemaps, see CO-WY
+        case ("sitemaps", true) => ???
+        case _ =>
+          val msg = s"Capability $capability not   supported."
+          throw new IllegalArgumentException(msg)
+      }
+  }
+
+  // Get responses according to the verb.
+  def getResponses: RDD[RsResponse] = {
+      records
+  }
```

Comment:
I left both methods because (I thought but didn't document) supporting a variety of capabilities would likely mean changes to at least getResponses() (I really tried to model this after your work on the OAI harvester, see below).   But since its unimplemented, I agree that it looks sketchy. 

https://github.com/dpla/ingestion3/blob/c92f935c7331a9aa32cc86c89ab133dea49e5aa8/src/main/scala/dpla/ingestion3/harvesters/oai/OaiRelation.scala#L92-L100

Reply:
There is probably some redundancy here that could be cleaned up.